### PR TITLE
Update NW-CLP pull through end of year

### PR DESCRIPTION
**Pull request by steeleb**
Hey KW! 

This looks like a lot of changes, but the take home is updating this branch to pull NW and CLP data through Dec 2023 and reflect changes in the lakeSR repository.

- high-level org changes: I eliminated tracking the _targets folder (and already deleted it in the upstream branch), now that we all know it does nothing for us to track the meta files other than add clutter. Moved the yml file to the root directory to increase readability.
- `data_acquisition/src/gee_functions.py` and `data_acquisition/src/runGEEperTile.py`: changes in these scripts are to reflect changes in lakeSR pipeline. Specifically, this grabs some additional pixel-level summaries for metadata relating to the thermal data (see [this issue](https://github.com/rossyndicate/NW-CLP-RS/issues)) and to add a count of all DSWE pixels greater than 0 (aka, water of some sort). The additional thermal metadata is part of our attempt to develop a better way to eliminate thermal outliers. The add of DSWE >0 is in part reality check, in part to satisfy the Loire project, if/when we return to it. Changes here are superficial only, adding info to the summary output.
- `data_acquisition/src/general_functions.R` changes here reflect a {targets} update that will not allow for tar_read() in a function - just moving the reference of the tar_read to the argument for function to satisfy the {targets} monsters.

I'd say look over to make sure everything tracks with the above descriptions, but I don't think this needs a super thorough look because most has been reviewed in lakeSR. 

Additional note: I know I need to pull this branch into the NW-CLP-RS repo as a step between a and b groups, but it's not going to happen right now. [Made an issue for it](https://github.com/rossyndicate/NW-CLP-RS/issues/8) in that repo, tho. When I move it over there, I'll take care of some of the formatting (like single quotes to double quotes and equals vs <- assignments which are probably all jumbled up here). Reasoning behind leaving it here is that there are lots of pulls that use this code, and it's not quite stable yet (see also [this issue](https://github.com/AquaSat/lakeSR/issues/19)).

Don't take more than 15 mins on this. If it takes longer than that, let me know and I'll assign elsewhere.

### CODE REVIEW

File: run_targets.Rmd [hunk 1]
```diff
@@ -9,12 +11,21 @@ output:
 # Purpose
 
 This script loads all necessary packages to run the targets pipeline, then runs it!
+This particular instance pulls the historical Landsat record for Northern Water and
+all reservoirs greater than 0.01km^2 in the Cache La Poudre HUC 8 watershed. The
+locations file referenced in the yaml config file was generated in targets branch
+a of the NW-CLP-RS repository.
```
Comment (kathryn-willi) [inline run_targets.Rmd]: add a link?

File: run_targets.Rmd [hunk 2]
```diff
@@ -9,12 +11,21 @@ output:
 # Purpose
 
 This script loads all necessary packages to run the targets pipeline, then runs it!
+This particular instance pulls the historical Landsat record for Northern Water and
+all reservoirs greater than 0.01km^2 in the Cache La Poudre HUC 8 watershed. The
+locations file referenced in the yaml config file was generated in targets branch
+a of the NW-CLP-RS repository. 
+
+Running this pipeline requires access to both the successfully run a branch of the
```
Comment (kathryn-willi) [inline run_targets.Rmd]:
```suggestion
Running this pipeline requires access to both the successfully run "a" branch of the
```

Comment (kathryn-willi) [reply run_targets.Rmd]: without the quotes i thought the sentence was funky, even though it was right

File: data_acquisition/src/gee_functions.py [hunk 3]
```diff
@@ -352,41 +353,47 @@ def ref_pull_457_DSWE1(image):
...
+    .combine(ee.Reducer.min().unweighted().forEachBand(pixOut.select(['min_SurfaceTemp', 'min_cloud_dist'])), sharedInputs = False)
     .combine(ee.Reducer.stdDev().unweighted().forEachBand(pixOut.select(['sd_Blue', 'sd_Green', 'sd_Red', 'sd_Nir', 'sd_Swir1', 'sd_Swir2', 'sd_SurfaceTemp'])), sharedInputs = False)
-    .combine(ee.Reducer.mean().unweighted().forEachBand(pixOut.select(['mean_Blue', 'mean_Green', 'mean_Red', 'mean_Nir', 'mean_Swir1', 'mean_Swir2', 'mean_SurfaceTemp', 'mean_temp_qa', 'mean_cloud_dist'])), sharedInputs = False)
+    .combine(ee.Reducer.mean().unweighted().forEachBand(pixOut.select(['mean_Blue', 'mean_Green', 'mean_Red', 
+              'mean_Nir', 'mean_Swir1', 'mean_Swir2', 'mean_SurfaceTemp'])), sharedInputs = False)
     .combine(ee.Reducer.kurtosis().unweighted().forEachBand(pixOut.select(['SurfaceTemp'])), outputPrefix = 'kurt_', sharedInputs = False)
-    .combine(ee.Reducer.count().unweighted().forEachBand(pixOut.select(['dswe1', 'dswe3'])), outputPrefix = 'pCount_', sharedInputs = False)
+    .combine(ee.Reducer.count().unweighted().forEachBand(pixOut.select(['dswe_gt0', 'dswe1', 'dswe3'])), outputPrefix = 'pCount_', sharedInputs = False)
```
Comment (kathryn-willi) [inline data_acquisition/src/gee_functions.py:396]: dswe_gt0 = total pixel count?

**Comment by kathryn-willi** [comment]
Don't see anything suspicious in here :) Good to merge

**Review by kathryn-willi — APPROVED** [review]

File: run_targets.Rmd (hunk 1 above)
Comment (steeleb) [reply run_targets.Rmd]: Absolutely.

File: data_acquisition/src/gee_functions.py (hunk 3 above)
Comment (steeleb) [reply data_acquisition/src/gee_functions.py:396]: that's right!! dswe greater than 0 is water of any type!

**Comment by steeleb** [comment]
Thank you Katie!

**Pull request merged**